internal/testhelpers/e2e_server.go

```diff
@@ -62,7 +82,70 @@ func StartE2EServer(t *testing.T, configFile string) (publicUrl, adminUrl string
 			return fmt.Errorf("expected status code 200 but got: %d", res.StatusCode)
 		}
 		return nil
-	}), err)
+	}))
+
+	return
+}
+
+func CheckE2EServerOnHTTPS(t *testing.T, publicPort, adminPort int) (publicUrl, adminUrl string) {
+	publicUrl = fmt.Sprintf("https://127.0.0.1:%d", publicPort)
+	adminUrl = fmt.Sprintf("https://127.0.0.1:%d", adminPort)
+
+	require.NoError(t, retry.Do(func() error {
+		tr := &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		}
+		client := &http.Client{Transport: tr}
+		res, err := client.Get(publicUrl + "/health/alive")
```

Comment:
@aeneasr isn't the health check endpoint be available in Admin API only?

Reply:
No, it's available in both :)

Reply:
ok. I check both APIs in HTTPS check